src/RTCRtpTransceiver.ts

```diff
@@ -0,0 +1,127 @@
+import { defineCustomEventTarget } from 'event-target-shim';
+import { NativeModules } from 'react-native';
+import { addListener } from './EventEmitter';
+import RTCRtpSender from './RTCRtpSender';
+import RTCErrorEvent from './RTCErrorEvent';
+import RTCRtpReceiver from './RTCRtpReceiver';
+
+const {WebRTCModule} = NativeModules;
+
+const TRANSCEIVER_EVENTS = [
+    'error'
+];
+
+export default class RTCRtpTransceiver extends defineCustomEventTarget(...TRANSCEIVER_EVENTS) {
+    _peerConnectionId: number;
+    _sender: RTCRtpSender;
+    _receiver: RTCRtpReceiver;
+
+    _id: string;
+    _mid: string | null;
+    _direction: string;
+    _currentDirection: string | null = null;
+    _stopped: boolean;
+
+    constructor(args: {
+        peerConnectionId: number,
+        id: string,
+        isStopped: boolean,
+        direction: string,
+        currentDirection: string,
+        mid?: string,
+        sender: RTCRtpSender,
+        receiver: RTCRtpReceiver,
+    }) {
+        super();
+        this._peerConnectionId = args.peerConnectionId;
+        this._id = args.id;
+        this._mid = args.mid ? args.mid : null;
+        this._direction = args.direction;
+        this._currentDirection = args.currentDirection;
+        this._stopped = args.isStopped;
+        this._sender = new RTCRtpSender({ peerConnectionId: args.peerConnectionId, id: args.sender.id, track: args.sender.track? args.sender.track : undefined });
```

Comment:
Isn't this done outside and we pass the sender / receiver object in the constructor? If not, why not?

Reply:
You are right.